challenge.rb

```diff
@@ -1,26 +1,26 @@
 
 def capitalize_each_string(input)
-  #implement your solution here
+  input.map{|x| x.capitalize}
 end
 
 def fetch_the_dog(input)
-  #implement your solution here
+  input.select{|x| x == "dog"}
 end
 
 def no_dogs_allowed(input)
-  #implement your solution here
+  input.reject {|x| x == "dog"}
 end
 
 def count_the_animals(input)
-  #implement your solution here
+	input.uniq.count
 end
 
 def fetch_the_first_two(input)
-  #implement your solution here
+	input.take(2)
 end
 
 def fetch_CD_animals(input)
-  #implement your solution here
+	input.find_all {|x| x[0] == "c" || x[0] == "d"}
```

Comment:
I kinda like [String#start_with?](http://ruby-doc.org/core-2.0.0/String.html#method-i-start_with-3F), but this solution is pretty clear too.